cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -655,6 +659,7 @@ public void programSpecification(HttpRequest request, HttpResponder responder,
                                    @PathParam("program-type") String type,
                                    @PathParam("program-name") String programName) throws Exception {
     ProgramType programType = getProgramType(type);
+    appVersion = replaceDefaultVersionWithLatest(namespaceId, appName, appVersion);
```

Comment:
@sumengwang should we introduce this change in this PR? I feel its doing only half the job? Lets be more organized and move [CDAP-19775] changes to a separate PR - it will be easier to track, in case there are bugs?